src/main/java/mate/academy/security/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package mate.academy.security.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.security.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    private static final int PASSWORD_MIN_LENGTH = 8;
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userEmail = userService.findByEmail(email);
+        User user = userEmail.get();
```

Comment:
This may be cause of NPE!
delete all needless and put in the if block something like:
userFromDB.isPresent() & userFromDB.get().getPassword()
.equals(HashUtil.hashPassword(password, userFromDB.get().getSalt()))